pkg/driver/node/node.go

```diff
@@ -58,28 +58,15 @@ var (
 
 // S3NodeServer is the implementation of the csi.NodeServer interface
 type S3NodeServer struct {
-	NodeID             string
-	Mounter            mounter.Mounter
-	credentialProvider *mounter.CredentialProvider
+	NodeID  string
+	Mounter mounter.Mounter
 }
 
-func NewS3NodeServer(nodeID string, mounter mounter.Mounter, credentialProvider *mounter.CredentialProvider) *S3NodeServer {
-	return &S3NodeServer{NodeID: nodeID, Mounter: mounter, credentialProvider: credentialProvider}
+func NewS3NodeServer(nodeID string, mounter mounter.Mounter) *S3NodeServer {
+	return &S3NodeServer{NodeID: nodeID, Mounter: mounter}
 }
 
 func (ns *S3NodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolumeRequest) (*csi.NodeStageVolumeResponse, error) {
-	volumeCtx := req.GetVolumeContext()
-	if volumeCtx[volumecontext.AuthenticationSource] == mounter.AuthenticationSourcePod {
-		podID := volumeCtx[volumecontext.CSIPodUID]
-		volumeID := req.GetVolumeId()
-		if podID != "" && volumeID != "" {
-			err := ns.credentialProvider.CleanupToken(volumeID, podID)
-			if err != nil {
-				klog.V(4).Infof("NodeStageVolume: Failed to cleanup token for pod/volume %s/%s: %v", podID, volumeID, err)
-			}
-		}
-	}
-
```

Comment:
This `NodeStageVolume` wasn't enabled at all – we need to add `STAGE_UNSTAGE_VOLUME` node capability to enable – and also this cleaning of tokens has been moved to `Unmount` method of mounter.